server/src/controllers/orderController.service.ts

```diff
@@ -39,6 +42,66 @@ export const ticketingWebhook = async (
   }
 };
 
+export const updateRefundStatus = async (
+  prisma: ExtendedPrismaClient,
+  paymentIntent: string,
+) => {
+  const order = await prisma.orders.findUnique({
+    where: {
+      payment_intent: paymentIntent,
+    },
+    select: {
+      orderid: true,
+    }
+  });
+
+  if (order === null) {
+    // there is no forseeable case where this should happen
+    console.error('received refund for order that does not exist with Payment Intent ' + paymentIntent);
+    return;
+  }
+
+  // Not all refund webhook events come with the correspodning refund object, but they do all contain
+  // the payment intent. The WonderTix schema allows multiple partial refunds, meaning a payment intent
+  // could have multiple refund intents. Thus, all refund intents connected to the payment intent are retrieved from
+  // the database and then have their status's checked by polling Stripe.
+  let refunds = await prisma.refunds.findMany({
+    where: {
+      orderid_fk: order.orderid,
+    },
+    select: {
+      id: true,
+      refund_intent: true,
+      refund_status: true,
+    },
+  });
+
+  refunds.forEach(async (refund) => {
```

Comment:
I think we generally want to avoid using forEach with async calls it might be worth swapping this for a for..of loop.

Reply:
I didn't know that, thanks! [Looks like map would allow for this part to be done in parallel, so I went with that.](https://stackoverflow.com/a/37576787)